src/zarr/core/array.py

```diff
@@ -2881,6 +2881,14 @@ def set_coordinate_selection(
         if hasattr(value, "shape") and len(value.shape) > 1:
             value = np.array(value).reshape(-1)
 
+        if not is_scalar(value, self.dtype) and (
+            isinstance(value, np.ndarray) and indexer.shape != value.shape
```

Comment:
I guess this would need to check for `is_ndarray_like`

Reply:
done, thanks!